gemrb/core/System/String.cpp

```diff
@@ -109,10 +109,12 @@ String* StringFromCString(const char* string)
 
 char* MBCStringFromString(const String& string)
 {
-	char* cStr = (char*)malloc(string.length()+1);
+	size_t allocatedBytes = string.length() * sizeof(String::value_type);
+	char *cStr = (char*)malloc(allocatedBytes);
```

Comment:
There is no actual need to do this here, `wcstombs` guarantees not to exceed the number of  bytes given and returns the actual bytes written that we increase by 1 and null afterwards. As nothing relies on finding NULL bytes here, I consider this safe. Made no sense before as well (no nulling), the lines at the bottom of this function were already sufficient.